src/sentry/static/sentry/app/views/projects/projectContext.jsx

```diff
@@ -193,11 +195,31 @@ const ProjectContext = createReactClass({
         errorType: ERROR_TYPES.MISSING_MEMBERSHIP,
       });
     } else {
-      this.setState({
-        loading: false,
-        error: true,
-        errorType: ERROR_TYPES.PROJECT_NOT_FOUND,
-      });
+      // The project may have been renamed, attempt to lookup the project, if
+      // we 301 we will recieve the moved project slug and can update update
+      // our route accordingly.
+      const lookupHandler = resp => {
+        const {status, responseJSON} = resp;
+
+        if (status !== 301 || !responseJSON || !responseJSON.detail) {
+          this.setState({
+            loading: false,
+            error: true,
+            errorType: ERROR_TYPES.PROJECT_NOT_FOUND,
+          });
+          return;
+        }
+
+        browserHistory.replace(
+          recreateRoute('', {
+            ...this.props,
+            params: {...this.props.params, projectId: responseJSON.detail.slug},
+          })
+        );
+      };
+
+      // The request ill 404 or 301
+      this.api.request(`/projects/${orgId}/${projectId}/`, {error: lookupHandler});
```

Comment:
Should this be here?

Reply:
Yeah, it calls the `lookupHandler` above (maybe style I could have inlined it, but I didn't like the indention) which is the thing that will 404 or 301, 301 indicating that the project was moved.

Reply:
Ah I see,  was confused with the nesting elsewhere: https://github.com/getsentry/sentry/pull/7766/files#diff-91d5b338b5c59f74442e617f2b3f56f0R151

Reply:
Yeah, it won't even make that call since the project slug isn't in the org list